File: testdata/jennies/builders/nullable_map_assignment/TypescriptBuilder/src/nullable_map_assignment/somestruct_builder_gen.ts
import * as cog from '../cog';
import * as nullable_map_assignment from '../nullable_map_assignment';

export class SomeStructBuilder implements cog.Builder<nullable_map_assignment.SomeStruct> {
    private readonly internal: nullable_map_assignment.SomeStruct;

    constructor() {
        this.internal = nullable_map_assignment.defaultSomeStruct();
    }

    build(): nullable_map_assignment.SomeStruct {
        return this.internal;
    }

    config(config: Record<string, string>): this {
        this.internal.config = config;
        return this;
    }
}

File: testdata/jennies/builders/nullable_map_assignment/GoBuilder/nullable_map_assignment/somestruct_builder_gen.go
package nullable_map_assignment

import (
	cog "github.com/grafana/cog/generated/cog"
)

var _ cog.Builder[SomeStruct] = (*SomeStructBuilder)(nil)

type SomeStructBuilder struct {
    internal *SomeStruct
    errors cog.BuildErrors
}

func NewSomeStructBuilder() *SomeStructBuilder {
	resource := NewSomeStruct()
	builder := &SomeStructBuilder{
		internal: resource,
		errors: make(cog.BuildErrors, 0),
	}

	return builder
}



func (builder *SomeStructBuilder) Build() (SomeStruct, error) {
	if err := builder.internal.Validate(); err != nil {
		return SomeStruct{}, err
	}
	
	if len(builder.errors) > 0 {
	    return SomeStruct{}, cog.MakeBuildErrors("nullable_map_assignment.someStruct", builder.errors)
	}

	return *builder.internal, nil
}

func (builder *SomeStructBuilder) Config(config map[string]string) *SomeStructBuilder {
    builder.internal.Config = config

    return builder
}

